### ISSUE TITLE
Error when i want to host EWTEAM web project.

### ISSUE THREAD

**Issue by adnanmerrakchi** [pc2ccs/pc2v9#560]
Hello All,

i tried to work with with the web client team [EWTeam]:
versions:
pc2 version: i tried  both pc2-9.5.4, pc2-9.6.0
OS : linux ubuntu server [nogui] 22.04.
browser: google chrome.

i followed all the steps in the file [setup.txt] but when i run the command **../bin/ewrun**
i get this error.

**Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /var/www/html/pc2team/lib/Java.inc:1978 Stack trace: #0 /var/www/html/pc2team/Login/login.php(21): Java->__call() #1 {main} thrown in /var/www/html/pc2team/lib/Java.inc on line 1978**

 please help!!! i have to get a stable version this weekend. thank you!!

**Comment by lane55** [comment]
One idea is that the server and the ewu pc2 versions are different.

**Comment by clevengr** [comment]
One issue is that you are using very old versions of both PC2 and EWTeam.  The current stable version of PC2 is 9.7.0, and there are Release Candidates for version 9.8.0, all available at https://pc2ccs.github.io/.    

The error you are seeing is related to problems with the PHP code in your EWTeam's web server.  The EWTeam web client has been deprecated and superseded by a completely new tool called the Web Team Interface (WTI).  This tool does the same thing as the old EWTeam, but it has its own internal web browser which completely eliminates the need to install PHP code in (or even use) an existing web server.  

The WTI is automatically downloaded as part of either PC2 V9.7 or V9.8, and there is a complete appendix in the PC2 Contest Administrator's Guide (under the **/doc** folder) explaining how to set up and use the WTI.  We strongly recommend that you try Version 9.7 (or 9.8) and the new WTI -- you'll be much more likely to have success.

**Comment by adnanmerrakchi** [comment]
Thank you so much for your response 
Hello @lane55 yes i used the same version, but now im trying to work with WTI.
Hello the big @clevengr yes i discoverd the WTI 2 days ago, and i succeed to deploy it.
But now i have a problem when i do some code mofidication in WTI-UI and i try to build the project by going to the WTI-API folder and run build with command **ant buildWTI**  it gave error **Buildfile: build.xml does not exist!** also there is a confusion when i check the the doc in **WTI-API/doc/TechDocs/WTI Installation Guide.pdf** the command is **ant build** and not **ant buildWTI** but anyway it's not working for both command (i tied to  build even before doing any code changes.).
please if you can help i will appreciate that i have a contest live this saturday and i am the system admin. i have to make everything done friday.
thank you so much

**Comment by clevengr** [comment]
If you read the file **README.TXT** in the WTI-API project folder, you will see that it says that the documentation you've been looking at is from the original student project which formed the basis of the WTI, and that the documentation there is out of date.  

Like most things related to PC2, the best (most current) source of information is on the PC2 Wiki, at https://github.com/pc2ccs/pc2v9/wiki.  Specifically, the page you want (information on building the WTI) is at https://github.com/pc2ccs/pc2v9/wiki/WTI-Build-Package-Details.

**Comment by adnanmerrakchi** [comment]
@clevengr  @lane55 yes i read the wiki and i ran the command **ant -buildfile packageWTI.xml** and :( it gives this error 

compile:
    [mkdir] Created dir: C:\Users\Adnan MERRAKCHI\Desktop\PC2\Github repo\pc2v9\projects\WTI-API\bin
    [javac] C:\Users\Adnan MERRAKCHI\Desktop\PC2\Github repo\pc2v9\projects\WTI-API\buildWTI.xml:74: warning: 'includeantruntime' was not set, defaulting to build.sysclasspath=last; set to false for repeatable builds
    [javac] Compiling 48 source files to C:\Users\Adnan MERRAKCHI\Desktop\PC2\Github repo\pc2v9\projects\WTI-API\bin
    [javac] Ignoring source, target and bootclasspath as release has been set
    [javac] C:\Users\Adnan MERRAKCHI\Desktop\PC2\Github repo\pc2v9\projects\WTI-API\src\main\controllers\ContestController.java:556: error: cannot find symbol
    [javac]                                     IInternalContest internalContest = scoreboardServerConn.getContest().getInternalContest();

**Comment by lane55** [comment]
The wiki ant build command is wrong.

**Comment by clevengr** [comment]
Since this issue originally referred to the EWTeam project, which is no longer supported, and the OP subsequently indicated he's switched to the WTI, I'm closing the issue as "Won't Fix".  If the OP still has problems with the WTI, please submit a new Issue and we'll be happy to look at it.
